### ISSUE TITLE
OCR fails if image is rotated

### ISSUE THREAD

**Issue by robertknight** [robertknight/tesseract-wasm#29]
OCR completely fails if the image is rotated at 90, 180 or 270 degrees. Tesseract has built-in orientation detection, so this could be used to resolve that.

**Comment by robertknight** [comment]
Tesseract's built-in orientation detection requires the library to be build with the legacy / non-LSTM text recognition engine. Leptonica has some built-in orientation detection functionality. So some options:

1. Compile Tesseract with the legacy engine included, so its orientation detection can be used
2. Use Leptonica's orientation detection
3. Don't support orientation detection and leave it as a problem for the consumer

**Comment by robertknight** [comment]
Work in progress at https://github.com/robertknight/tesseract-wasm/pull/34.

**Comment by robertknight** [comment]
https://github.com/robertknight/tesseract-wasm/pull/34 adds a partial solution in the form of orientation detection, however the algorithm is simplistic and this means that in any application user input would probably be required to confirm actions depending on it.

**Comment by robertknight** [comment]
I posted a [comment](https://news.ycombinator.com/item?id=32032839) on Hacker News and someone responded with a test case where the word recognition works well, but the text is not output in the correct order, due rotation of the image:

![C1jn2Kz](https://user-images.githubusercontent.com/2458/178115153-359453f0-07fa-4608-9ff9-46966d66b634.png)

If you compare the text output of this image in the demo, vs a copy of this image rotated such that the text baselines are straight, you can see that the layout outputs are different.

**Comment by johanvaneck** [comment]
Any updates on this?

**Comment by robertknight** [comment]
No. Ensuring the input is correctly oriented is currently a problem that users of the library have to solve.
